### PR TITLE
[WIP] BLSCT Proof of Stake

### PR DESCRIPTION
**Pull request by aguycalled**

### CODE REVIEW

File: src/blsct/range_proof/generators.h [hunk 1]
```diff
@@ -69,7 +69,7 @@ class GeneratorsFactory
 
 private:
     inline const static GeneratorDeriver m_deriver =
-        GeneratorDeriver<typename T::Point>("bulletproofs");
+        GeneratorDeriver<typename T::Point>("proof-of-stake");
```
Comment (gogoex) [inline src/blsct/range_proof/generators.h:72]: The intension may be clearer if we make "proof-of-stake" a constant and use it in `set_mem_proof_setup.h` as well?

**Review by gogoex — APPROVED** [review]
LGTM!

**Comment by aguycalled** [comment]
> What should I be testing with the code in this PR?

I would review the code, and test as indicated in the article. Using the `blsctregtest` network, run a few nodes locally with their respective stakers.

**Comment by mxaddict** [comment]
Alrighty, already running some nodes locally, have 3 of them staking with
different amounts, what should the block time/reward be?


Cheers,
Barry Deeney

**Pull request merged**